### PR TITLE
Add workaround for maintenance mode

### PR DESCRIPTION
**Pull request by bk201**
Related issue: https://github.com/harvester/harvester/issues/7128

### CODE REVIEW

**Comment by bk201** [comment]
To add all versions after approval.

**Review by WebberHuang1118 — APPROVED** [review]
LGTM, thanks.

**Review by ibrokethecloud — APPROVED** [review]

File: docs/host/host.md [hunk 1]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
 
 ## Node Maintenance
 
+
+:::warning
+
+We found that there is a chance that VMs on a node can encounter an I/O error while the node enters maintenance mode. Please follow this procedure to avoid the problem before a fix is released:
```
Comment (ihcsim) [inline docs/host/host.md:27]: Meanwhile, how does a user recover from this? Would taking the node out of maintenance mode, add the taint, re-enter maintenance mode work?

File: docs/host/host.md [hunk 2]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+
+- Set the taint on the maintenance node:
+    ```
+    kubectl taint node <NODE> --overwrite kubevirt.io/drain=draining:NoSchedule
+    ```
+- Wait for all VMs to be live-migrated out of the maintenance node.
+- Select **Enable Maintenence Mode** in the GUI.
```
Comment (ihcsim) [inline docs/host/host.md:34]:
```suggestion
- Select **Enable Maintenance Mode** in the GUI.
```

File: docs/host/host.md (hunk 1 above)
Comment (ihcsim) [inline docs/host/host.md:27]:
```suggestion
A recent [bug](https://github.com/harvester/harvester/issues/7128) reveals that there is a chance that VMs on a node can encounter an I/O error while in maintenance mode. Users [must|should] follow the following maintenance procedure until a fix is released:
```
I think if this is something that we want every user to follow, we should be more explicit about it.

Comment (bk201) [reply docs/host/host.md:27]: If a VM enters an I/O error, the only way to recover it is to stop it first and start it again. Let me add this.

Comment (bk201) [reply docs/host/host.md]: Updated in the issue description (somehow I think it's a better place).

File: docs/host/host.md [hunk 3]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
 
 ## Node Maintenance
 
+
+:::warning
+
+A recent [bug](https://github.com/harvester/harvester/issues/7128) reveals that there is a chance that VMs on a node can encounter an I/O error while in maintenance mode. Users should follow the following maintenance procedure until a fix is released:
```
Comment (LucasSaintarbor) [inline docs/host/host.md:27]:
```suggestion
Due to a recent [bug](https://github.com/harvester/harvester/issues/7128), there is a possibility that virtual machines (VMs) on a node may encounter an I/O error while in maintenance mode. To mitigate this issue until a fix is released, we advise using the following maintenance procedure:
```

File: docs/host/host.md [hunk 4]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+
+- Set the taint on the maintenance node:
```
Comment (LucasSaintarbor) [inline docs/host/host.md:29]:
```suggestion
1. Set the taint on the maintenance node:

```

File: docs/host/host.md [hunk 5]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+    ```
```
Comment (LucasSaintarbor) [inline docs/host/host.md:30]:
```suggestion
    ```sh
```

File: docs/host/host.md [hunk 6]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+    kubectl taint node <NODE> --overwrite kubevirt.io/drain=draining:NoSchedule
+    ```
+- Wait for all VMs to be live-migrated out of the maintenance node.
```
Comment (LucasSaintarbor) [inline docs/host/host.md:33]:
```suggestion

1. Wait for all VMs to be live-migrated out of the maintenance node.
```

File: docs/host/host.md [hunk 7]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+- Select **Enable Maintenance Mode** in the GUI.
```
Comment (LucasSaintarbor) [inline docs/host/host.md:34]:
```suggestion
1. Select **Enable Maintenance Mode** in the UI.
```

File: docs/host/host.md [hunk 8]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+
+Once you finish the maintenance work, you need to:
```
Comment (LucasSaintarbor) [inline docs/host/host.md:36]:
```suggestion
After finishing the maintenance work, you need to:
```

File: docs/host/host.md [hunk 9]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+- Remove the taint on the maintenance node:
```
Comment (LucasSaintarbor) [inline docs/host/host.md:37]:
```suggestion
1. Remove the taint on the maintenance node:

```

File: docs/host/host.md [hunk 10]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+- Set the taint on the maintenance node:
+    ```
+    kubectl taint node <NODE> --overwrite kubevirt.io/drain=draining:NoSchedule
+    ```
+- Wait for all VMs to be live-migrated out of the maintenance node.
+- Select **Enable Maintenance Mode** in the GUI.
+
+Once you finish the maintenance work, you need to:
+- Remove the taint on the maintenance node: 
+    ```
```
Comment (LucasSaintarbor) [inline docs/host/host.md:38]:
```suggestion
    ```sh
```

File: docs/host/host.md [hunk 11]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+    kubectl taint node <NODE> kubevirt.io/drain-
+    ```
+- Select **Disable Maintenence Mode** in the GUI.
```
Comment (LucasSaintarbor) [inline docs/host/host.md:41]:
```suggestion

1. Select **Disable Maintenance Mode** in the UI.
```

File: docs/host/host.md [hunk 12]
```diff
@@ -21,6 +21,28 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+
+For more information, please check the [issue](https://github.com/harvester/harvester/issues/7128).
```
Comment (LucasSaintarbor) [inline docs/host/host.md:43]:
```suggestion
For more information, please refer to issue [#7128](https://github.com/harvester/harvester/issues/7128).
```

**Review by LucasSaintarbor — CHANGES_REQUESTED** [review]
Made some formatting / grammar changes. Please lmk if these look okay. Should this content be added to the v1.4 docs? Thanks!

**Comment by bk201** [comment]
Thanks @LucasSaintarbor (I'll add all v1.x docs after everything looks good)

**Review by jillian-maroket — APPROVED** [review]

File: docs/host/host.md [hunk 13]
```diff
@@ -21,6 +21,32 @@ Because Harvester is built on top of Kubernetes and uses etcd as its database, t
...
+
+Due to a recent [bug](https://github.com/harvester/harvester/issues/7128), there is a possibility that virtual machines (VMs) on a node may encounter an I/O error while in maintenance mode. To mitigate this issue until a fix is released, we advise using the following maintenance procedure:
+
+1. Set the taint on the maintenance node:
+
+    ```sh
+    kubectl taint node <NODE> --overwrite kubevirt.io/drain=draining:NoSchedule
+    ```
+
+1. Wait for all VMs to be live-migrated out of the maintenance node.
```
Comment (innobead) [inline docs/host/host.md:35]: Shouldn't the order of these steps be reversed?

```
1. Set the taint on the maintenance node.
1. Select **Enable Maintenance Mode** in the UI.
1. Wait for all VMs to be live-migrated out of the maintenance node.
```

Comment (innobead) [reply docs/host/host.md:35]: @bk201 clarified this with me. All good. 

ref: 
- https://kubevirt.io/user-guide/cluster_admin/node_maintenance/#evict-all-vms-from-a-node
- https://github.com/harvester/harvester/pull/7133

**Pull request merged**